lib/mpl_toolkits/axisartist/axis_artist.py

```diff
@@ -156,12 +156,12 @@ def draw(self, renderer):
             join = self._solidjoinstyle
         gc.set_joinstyle(join)
         gc.set_capstyle(cap)
+        gc.set_dashes(self._dashOffset, self._dashSeq)
 
-        funcname = self._lineStyles.get(self._linestyle, '_draw_nothing')
```

Comment:
This will now raise on non-valid `self._linestyle`.  Do the git logs provide any clue why was this coded this way?